django/db/backends/postgresql/base.py

```diff
@@ -179,6 +197,54 @@ class DatabaseWrapper(BaseDatabaseWrapper):
     ops_class = DatabaseOperations
     # PostgreSQL backend-specific attributes.
     _named_cursor_idx = 0
+    _connection_pools = {}
+
+    @property
+    def pool(self):
+        pool_options = self.settings_dict["OPTIONS"].get("pool")
+        if not pool_options or self.alias == NO_DB_ALIAS or not is_psycopg3:
```

Comment:
Checking `is_psycopg3` seems redundant as it already raises `ImproperlyConfigured`.

Reply:
#17900